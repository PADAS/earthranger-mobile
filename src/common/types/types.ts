import { EventStatus, PatrolResult } from '../enums/enums';

export type Position = [number, number];

export type RootStackParamList = {
  SplashScreen: undefined,
  LoginView: undefined,
  TrackLocationMapView: {
    flyTo?: Position;
  }
  MainTabBar: undefined,
  ReportsView: undefined,
  ReportIssueView: undefined,
  PinAuthenticationView: undefined,
  SyncLoaderView: undefined,
  ReportCategoriesView: {
    coordinates?: Position,
  },
  ReportTypesView: {
    title: string,
    categoryId: string,
    coordinates?: Position,
    isPatrolInfoEventType?: boolean,
  },
  ReportForm: {
    reportId?: number,
    title: string,
    typeId: string,
    coordinates?: Position
    schema: string,
    geometryType: string,
    isEditMode?: boolean,
    createdAt?: string,
    isDefaultPatrolInfoEnabled?: boolean,
    geoJson?: GeoJSON.FeatureCollection,
    areaMeters?: number,
    perimeterMeters?: number,
    polygonPoints?: string,
    viewBox?: string,
    isPatrolInfoEvent?: boolean,
  },
  ReportEditLocationView: { coordinates: Position },
  RecordReportAreaView: undefined,
  AboutView: undefined,
  ReportNoteView: { id: number, text: string, coordinates: Position, channelId: string },
  MenuSettingsView: undefined,
  SettingsView: undefined,
  StatusView: undefined,
  ReportRepeatableFieldListView: {
    schema: any,
    title: string,
    rendererId: string,
    data?: RepeatableFieldListData[],
  },
  RepeatableFormView: {
    schema: any,
    title: string,
    listId: string,
  },
  EventsListFilterView: undefined,
  ReportsPendingSync: undefined,
  TrackedBySubjectsView: {
    subjects?: any,
  },
  BasemapView: undefined,
  CoordinateUnitsView: undefined,
  PatrolsTypeView: {
    coordinates: Position,
  },
  StartPatrolView: {
    patrolTitle: string,
    patrolType: string,
    patrolTypeId: string,
    coordinates: Position,
  },
  PhotoQualityView: undefined,
  ResetDatabaseCacheView: undefined,
  SubjectsListView: {
    parentSubjectGroupName?: string,
    isParentView?: boolean,
    parentId?: string,
  }
};

export type BottomSheetParamList = {
  Basemap: undefined,
  PatrolDetailsView: { patrolId: number },
  PatrolTrack: undefined,
  TrackingStatusView: undefined,
  SubjectDetailsView: undefined,
};

export interface EdgeInsets {
  top: number;
  right: number;
  bottom: number;
  left: number;
}

export type RepeatableFieldListData = {
  id: number;
  title: string;
  data: any;
  hidden: boolean,
};

export type LastLocation = {
  timestamp: string;
  displayDate: string;
  latitude: number;
  longitude: number;
  accuracy: number;
};

export interface IconProps {
  width?: string;
  height?: string;
  color?: string;
  viewbox?: string;
  isDisabled?: boolean;
}

export type AttachmentPhotoType = 'photo';
export type AttachmentNoteType = 'note';
export type AttachmentUpdateNoteType = 'updateNote';
export type AttachmentAddNewNoteType = 'addNewNote';
export type AttachmentDeleteNoteType = 'deleteNote';

export type ReportFormEvent = {
  id: number;
  remoteId: string,
  accountId: string,
  eventTypeId: number;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string;
  event_values: string;
  isDraft: number;
  createdAt: number,
  updatedAt: number,
  patrol_segment_id: number | undefined,
};

export type EventData = {
  eventTypeId: string;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string;
  event_values: string;
  isDraft: number;
  createdAt: string,
  updatedAt: string,
  patrol_segment_id: number | undefined,
};

export type ReportFormFullEvent = {
  account_id: number;
  profile_id: number;
  event_type_id: string;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string,
  event_values: string;
  is_draft: number;
  value: string;
  updated_at: string;
  created_at: string;
  id: string;
  geometry_type: string;
  segment_remote_id: string | undefined;
  patrol_segment_id: string | undefined;
};

export type PersistedEventType = {
  id?: number;
  remote_id: string;
  account_id: number;
  profile_id: string;
  value: string;
  display: string;
  schema: string;
  category_id: string;
  geometry_type: string;
  default_priority: string;
  icon: string;
  icon_svg: string;
};

export type PersistedPatrolType = {
  id: number;
  remote_id: string;
  account_id: number;
  value: string;
  display: string;
  icon: string;
  icon_svg: string;
  default_priority: number;
  is_active: number;
  is_selected: boolean;
};

export type PersistedPatrol = {
  id: number;
  remote_id: string;
  account_id: number;
  profile_id: number;
  title: string;
  priority: string;
  state: string;
  created_at: string;
  updated_at: string;
};

export type PersistedPatrolSegment = {
  id: number
  remote_id: string;
  patrol_id: number;
  patrol_type_id: number;
  start_latitude: string;
  start_longitude: string;
  stop_latitude: string;
  stop_longitude: string;
  start_time: string;
  end_time: string;
};

export type PersistedPatrolTypeSegment = {
  id: number;
  remote_id: string;
  patrol_id: number;
  patrol_type_id: number;
  start_latitude: string;
  start_longitude: string;
  stop_latitude: string;
  stop_longitude: string;
  start_time: string;
  end_time: string;
  patrol_type: string;
};

export type DraftReports = {
  id: number;
  title: string;
  type_id: number,
  updated_at: string;
  icon_svg: string;
  default_priority: number;
  schema: string,
  geometry_type: string,
  hidden?: boolean;
  is_draft?: boolean;
};

export type EventUploadDetails = {
  reportId: number;
  reportTitle: string;
  isReportUploaded: boolean;
  uploaded: {
    notes: number;
    images: number;
  };
  pending: {
    notes: number;
    images: number;
  };
  state: string;
};

export type ReportFormEventAttachments = {
  id: number,
  remote_id: string,
  profile_id: number,
  account_id: number,
  event_id: string;
  type: string;
  path: string;
  thumbnail_path: string,
  note_text: string;
  uploaded: number,
  attachment_id: string;
};

export type PersistedPatrolDetail = {
  remote_id: string,
  title: string,
  serial_number: string,
  start_latitude: number,
  start_longitude: number,
  start_time: number,
  icon_svg: number,
  created_at: number,
};

export type PatrolDetail = {
  remoteId: string,
  title: string,
  serialNumber: string,
  startLatitude: number,
  startLongitude: number,
  startTime: number,
  iconSVG: string
  startTimeFormatted: string,
  createdAt: number,
};

export type PersistedUserProfile = {
  id: number;
  remote_id: string;
  account_id: string;
  username: string;
  content_type: string;
  subject_id: string;
  pin: string;
  permissions: string,
};

export type UserInfo = {
  id: number,
  type: string,
  permissions: string,
};

export type PersistedUserSubject = {
  id: number;
  remote_id: string;
  account_id: string;
  name: string;
  content_type: string;
  isSelected: boolean;
};

export type Patrol = {
  id: number,
  typeId: string,
  type: string,
  display: string,
  coordinates: Position | null,
  status: PatrolResult,
  timestamp: string,
};

// Polygon Reducer

export type PolygonState = {
  past: Position[],
  current: Position[],
};

export type PolygonAction = {
  type: 'UNDO' | 'REDO' | 'ADD',
  value?: Position,
};

type TGeometry = 'polygon' | 'point';

export const Geometry: Record<TGeometry, string> = {
  polygon: 'polygon',
  point: 'point',
};

export type Size = {
  width: number,
  height: number,
};

export interface EventListItem {
  bgColor: string,
  defaultPriority: number,
  error: string,
  fgColor: string,
  hidden: boolean,
  icon: string,
  id: number,
  isDraft: boolean,
  labelText: string,
  remoteId: string,
  status: EventStatus,
  statusIcon: string,
  text: string,
  title: string,
  errorMsg?: string,
  hasErrorOnEvent: boolean,
}

export type SubjectGroupData = {
  count: number,
  isVisible: boolean,
  title: string,
  type: 'group',
  subjects: number[],
  id: string,
};

export type Subject = {
  id: number,
  name: string,
  type: 'subject',
  isVisible: boolean,
  icon: string,
  lastPosition: Position,
  lastPositionUpdate: string,
  isHidden: boolean,
};

export type SubjectGroupSyncItem = {
  name: string,
  remoteId: string,
  parentId?: string,
  parentLocalId?: string,
};

export type SubjectSyncItem = {
  iconSVG: string,
  isActive: string,
  lastPosition: string,
  lastPositionDate: string,
  name: string,
  remoteId: string,
  tracksAvailable: string,
  updatedAt: string,
};
