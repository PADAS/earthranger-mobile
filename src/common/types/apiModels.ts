export interface FileRequest {
  uri: string,
  type: string,
  name: string,
}

export enum ApiStatus {
  Unknown = 1,
  Succeeded = 200,
  BadRequest = 400,
  Unauthorized = 401,
  Forbidden = 403,
  NotFound = 404,
  ServerError = 500,
}
