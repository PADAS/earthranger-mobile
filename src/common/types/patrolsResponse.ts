import { Subject } from './usersResponse';

export interface Status {
  code: number;
  message: string;
}

export interface PatrolType {
  default_priority: number;
  display: string;
  icon_id: string;
  id: string;
  is_active: boolean;
  value: string;
  ordernum: number;
}

export interface PatrolTypesResponse {
  data: PatrolType[];
  status: Status;
}

export interface TrackPatrolByResponse {
  data: {
    properties: {
      leader: {
        enum: Subject [],
      }
    }
  }
  status: Status,
}
