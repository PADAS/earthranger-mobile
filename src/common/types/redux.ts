export interface InitialState {
  session: {
    accessToken: string,
    refreshToken: string,
    error: null,
    sessionError: null,
  },
  reports: {
    noteId: number,
    textNote: string,
  }
}

export type ActionType = 'REFRESH_TOKEN'
| 'REFRESH_TOKEN_ERROR'
| 'USER_LOGIN'
| 'USER_LOGIN_ERROR'
| 'GET_SESSION'
| 'GET_SESSION_ERROR'
| 'CREATE_NOTE'
| 'CLEAR_NOTE_DATA'
| 'NUMBER_OF_REPORT_DRAFTS'
| 'CLEAR_SESSION';

export interface ReducerAction {
  type: ActionType,
  payload: any;
}
