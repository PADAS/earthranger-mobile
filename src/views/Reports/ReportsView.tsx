// External Dependencies
import React, {
  useCallback, useEffect, useRef, useState,
} from 'react';
import { View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { useFocusEffect } from '@react-navigation/native';
import { useMMKVBoolean, useMMKVNumber } from 'react-native-mmkv';

// Internal Dependencies
import { AddEventButton } from '../../common/components/AddEventButton/AddEventButton';
import {
  IS_STATUS_FILTER_DRAFT_SELECTED,
  IS_STATUS_FILTER_PENDING_SELECTED,
  IS_STATUS_FILTER_SYNCED_SELECTED,
  MERGE_CATEGORIES_KEY,
  REPORTS_SUBMITTED_KEY,
  REPORTS_SYNCING,
  SESSION_KEY,
} from '../../common/constants/constants';
import { Position, RootStackParamList } from '../../common/types/types';
import {
  getBoolForKey,
  localStorage,
} from '../../common/data/storage/keyValue';
import {
  useRetrieveReportPendingSync,
} from '../../common/data/reports/useRetrieveReportPendingSync';
import { useUploadReports } from '../../common/data/reports/useUploadReports';
import {
  getSession,
} from '../../common/data/storage/session';
import { isInternetReachable } from '../../common/utils/NetworkInfoUtils';
import { getMapUserPosition } from '../../common/utils/locationUtils';
import { useRefreshToken } from '../../common/utils/useRefreshToken';
import { localStorageSecured } from '../../common/data/storage/utils';
import { ApiStatus } from '../../common/types/apiModels';
import AnalyticsEvent, { analyticsEventToHashMap } from '../../analytics/model/analyticsEvent';
import { logEvent } from '../../analytics/wrapper/analyticsWrapper';
import { addReportEvent } from '../../analytics/reports/reportsAnalytics';
import { isSyncingReports } from '../../common/utils/syncUtils';
import { EventsList } from './components/EventsList/EventsList';

// Styles
import styles from './ReportsView.styles';

// Interfaces + Types
interface ReportsViewProps {
  navigation: NativeStackNavigationProp<RootStackParamList, 'ReportsView'>;
}

const ReportsView = ({
  navigation,
}: ReportsViewProps) => {
  // Hooks
  const { retrieveReportPendingSyncCount } = useRetrieveReportPendingSync();
  const { uploadReportAndAttachments } = useUploadReports();
  const { handleRefreshToken } = useRefreshToken();

  // Components Reference
  const isUploadAllowed = useRef(true);
  const accessToken = useRef(getSession()?.access_token || '');

  // Component's State
  const [reportSyncStatusChanged] = useMMKVBoolean(REPORTS_SYNCING, localStorage);
  const [pendingSyncCount, setPendingSyncCount] = useState(-1);
  const [reportsSubmittedCount, setReportsSubmittedCount] = useMMKVNumber(
    REPORTS_SUBMITTED_KEY,
    localStorage,
  );

  useFocusEffect(useCallback(() => {
    async function syncStatusReportsAsync() {
      const internetAvailable = await isInternetReachable();
      if (!internetAvailable) {
        return;
      }
      if (isUploadAllowed.current && !isSyncingReports()) {
        isUploadAllowed.current = false;
        await syncStatusReports();
        isUploadAllowed.current = true;
      }
    }

    updateDashboard();

    if (pendingSyncCount > 0) {
      syncStatusReportsAsync();
    }
  }, [
    accessToken.current,
    pendingSyncCount,
    getBoolForKey(IS_STATUS_FILTER_DRAFT_SELECTED),
    getBoolForKey(IS_STATUS_FILTER_PENDING_SELECTED),
    getBoolForKey(IS_STATUS_FILTER_SYNCED_SELECTED),
  ]));

  useEffect(() => {
    if (!isSyncingReports()) {
      updateDashboard();
    }
  }, [reportSyncStatusChanged]);

  useEffect(() => {
    const valueChangeListener = localStorageSecured.addOnValueChangedListener((key: string) => {
      if (key === SESSION_KEY) {
        accessToken.current = getSession()?.access_token || '';
      }
    });

    return () => {
      valueChangeListener.remove();
    };
  }, []);

  const updateDashboard = async () => {
    await getPendingSyncCount();
  };

  const syncStatusReports = async () => {
    const response = await uploadReportAndAttachments();

    if (response.reportStatus === ApiStatus.Unauthorized
      || response.attachmentStatus === ApiStatus.Unauthorized) {
      await handleRefreshToken(navigation);
    } else if (response.reportStatus === ApiStatus.Succeeded
      || response.attachmentStatus === ApiStatus.Succeeded) {
      setReportsSubmittedCount(reportsSubmittedCount + pendingSyncCount);
      await getPendingSyncCount();
    }
  };

  // Handlers
  const onPressHandler = () => {
    trackAnalyticsEvent(addReportEvent());
    if (getBoolForKey(MERGE_CATEGORIES_KEY)) {
      navigation.navigate('ReportTypesView', {
        title: '',
        categoryId: '',
        coordinates: getMapUserPosition() as Position,
      });
    } else {
      navigation.navigate('ReportCategoriesView', { coordinates: getMapUserPosition() as Position });
    }
  };

  // Utility Functions
  const getPendingSyncCount = async () => {
    const count = await retrieveReportPendingSyncCount();
    setPendingSyncCount(count);
  };

  const trackAnalyticsEvent = useCallback(async (event: AnalyticsEvent) => {
    await logEvent(event.eventName, analyticsEventToHashMap(event));
  }, []);

  return (
    <SafeAreaView style={styles.containerSafeArea} edges={['bottom']}>
      {/* Main Container */}
      <EventsList
        navigation={navigation}
      />

      <View style={styles.addReportButton}>
        <AddEventButton onPress={onPressHandler} />
      </View>
      {/* End Main Container */}
    </SafeAreaView>
  );
};

export { ReportsView };
