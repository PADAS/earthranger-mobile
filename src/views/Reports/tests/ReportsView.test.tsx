// External Dependencies
import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import * as MMKV from 'react-native-mmkv';

// Internal Dependencies
import { ReportsView } from '../ReportsView';
import { RootStackParamList } from '../../../common/types/types';

// Types + Interfaces
// eslint-disable-next-line max-len
type StackNavigationPropAlias = NativeStackNavigationProp<RootStackParamList, 'ReportsView'>;

jest.mock('geodesy/latlon-ellipsoidal-vincenty', () => ({
  LatLon: jest.fn(),
}));

jest.mock('geodesy/utm', () => ({
  LatLon: jest.fn(),
}));

jest.mock('geodesy/mgrs', () => ({
  LatLon: jest.fn(),
}));

jest.mock('@react-navigation/native', () => ({
  useFocusEffect: jest.fn(),
}));

describe('ReportsView', () => {
  let navigation: Pick<StackNavigationPropAlias, 'navigate'>;

  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
      // @ts-ignore
      addListener: jest.fn().mockImplementation((event, callback) => {
        callback();
        return {
          remove: jest.fn(),
        };
      }),
    };
  });

  it('matches snapshot', () => {
    jest.spyOn(MMKV, 'useMMKVBoolean').mockImplementation(() => ([false, jest.fn()]));
    jest.spyOn(MMKV, 'useMMKVNumber').mockImplementation(() => ([1, jest.fn()]));
    const { toJSON } = render(
      <ReportsView navigation={navigation} reportDraftsCount={0} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('shows submitted info customAlert', async () => {
    jest.spyOn(MMKV, 'useMMKVBoolean').mockImplementation(() => ([false, jest.fn()]));
    jest.spyOn(MMKV, 'useMMKVNumber').mockImplementation(() => ([1, jest.fn()]));

    const { getByTestId, getByText } = render(
      <ReportsView navigation={navigation} reportsDraftCount={0} />,
    );
    await waitFor(() => {
      const showSubmittedInfoIcon = getByTestId('ReportsView-InfoIcon');
      fireEvent.press(showSubmittedInfoIcon);
    });

    expect(getByText('When the device has a good connection the app will sync submited reports and update changes to the report forms. The number of submitted reports will reset each time you switch active users.')).toBeTruthy();
  });
});
