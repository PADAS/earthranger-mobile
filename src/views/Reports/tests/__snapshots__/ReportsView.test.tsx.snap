// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportsView matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
    ]
  }
  style={
    Object {
      "backgroundColor": "#F9F9F9",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#DDDDDD",
                "borderRadius": 3,
                "borderWidth": 1,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 16,
                "paddingBottom": 28,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#63666A",
                  "fontSize": 13,
                  "fontWeight": "500",
                  "textTransform": "uppercase",
                }
              }
            >
              SAVED DRAFTS
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="26"
                  bbWidth="32"
                  fill="#B62879"
                  focusable={false}
                  height="26"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 26,
                        "width": 32,
                      },
                    ]
                  }
                  vbHeight={26}
                  vbWidth={32}
                  width="32"
                >
                  <RNSVGGroup
                    fill={4290127993}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M12.8 0H3.2C1.44 0 0.016 1.44 0.016 3.2L0 22.4C0 24.16 1.44 25.6 3.2 25.6H28.8C30.56 25.6 32 24.16 32 22.4V6.4C32 4.64 30.56 3.2 28.8 3.2H16L12.8 0Z"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 21,
                      "fontWeight": "400",
                      "letterSpacing": 0.02,
                      "marginLeft": 14,
                      "marginRight": 32,
                    }
                  }
                >
                  -1
                </Text>
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="24"
                  bbWidth="15"
                  fill="#63666A"
                  focusable={false}
                  height="24"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 15,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={15}
                  width="15"
                >
                  <RNSVGGroup
                    fill={4284704362}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M0 21.876L2.124 24L14.124 12L2.124 0L0 2.124L9.876 12L0 21.876Z"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#DDDDDD",
                "borderRadius": 3,
                "borderWidth": 1,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 16,
                "paddingBottom": 28,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#63666A",
                  "fontSize": 13,
                  "fontWeight": "500",
                  "textTransform": "uppercase",
                }
              }
            >
              PENDING SYNC
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="25"
                  bbWidth="32"
                  fill="#0056C7"
                  focusable={false}
                  height="25"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 25,
                        "width": 32,
                      },
                    ]
                  }
                  vbHeight={25}
                  vbWidth={32}
                  width="32"
                >
                  <RNSVGGroup
                    fill={4278212295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M25.8 8.1459C24.8933 3.49303 20.8533 0 16 0C14.0267 0 12.2 0.579923 10.6533 1.57793L12.6 3.54697C13.6133 3.00751 14.7733 2.69732 16 2.69732C20.0533 2.69732 23.3333 6.01502 23.3333 10.1149V10.7893H25.3333C27.5467 10.7893 29.3333 12.5965 29.3333 14.8352C29.3333 16.3592 28.48 17.6809 27.2533 18.3687L29.1867 20.3243C30.88 19.097 32 17.101 32 14.8352C32 11.2748 29.2667 8.38866 25.8 8.1459ZM4 1.7128L7.66667 5.40812C3.41333 5.59694 0 9.13042 0 13.4866C0 17.9507 3.58667 21.5785 8 21.5785H23.64L26.3067 24.2759L28 22.5631L5.69333 0L4 1.7128ZM10.3067 8.09195L20.9733 18.8812H8C5.05333 18.8812 2.66667 16.4671 2.66667 13.4866C2.66667 10.5061 5.05333 8.09195 8 8.09195H10.3067Z"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 21,
                      "fontWeight": "400",
                      "letterSpacing": 0.02,
                      "marginLeft": 14,
                      "marginRight": 32,
                    }
                  }
                >
                  -1
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#DDDDDD",
              "borderRadius": 3,
              "borderWidth": 1,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 16,
              "paddingBottom": 28,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#63666A",
                "fontSize": 13,
                "fontWeight": "500",
                "textTransform": "uppercase",
              }
            }
          >
            STATUS
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 8,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#63666A",
                      "fontSize": 15,
                    }
                  }
                >
                  Submitted This Session: 
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 15,
                      "fontWeight": "400",
                    }
                  }
                >
                  1
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#63666A",
                      "fontSize": 15,
                    }
                  }
                >
                  Last Sync: 
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 15,
                      "fontWeight": "400",
                    }
                  }
                >
                  Not yet synced
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-end",
              "paddingRight": 16,
              "paddingTop": 23,
            }
          }
          testID="ReportsView-InfoIcon"
        >
          <View>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="16"
              bbWidth="16"
              fill="#63666A"
              focusable={false}
              height="16"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
              vbHeight={16}
              vbWidth={16}
              width="16"
            >
              <RNSVGGroup
                fill={4284704362}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M8 0C3.584 0 0 3.584 0 8C0 12.416 3.584 16 8 16C12.416 16 16 12.416 16 8C16 3.584 12.416 0 8 0ZM8.8 12H7.2V7.2H8.8V12ZM8.8 5.6H7.2V4H8.8V5.6Z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 22,
        "position": "absolute",
        "right": 8,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0056C7",
          "borderRadius": 24,
          "elevation": 5,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "minWidth": undefined,
          "opacity": 1,
          "paddingHorizontal": undefined,
          "paddingVertical": undefined,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 2,
          "width": 48,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="22"
        bbWidth="22"
        focusable={false}
        height="22"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 22,
              "width": 22,
            },
          ]
        }
        vbHeight={22}
        vbWidth={22}
        width="22"
      >
        <RNSVGGroup>
          <RNSVGGroup
            clipPath="clip0_464_280"
          >
            <RNSVGPath
              d="M9.9 15.84H3.96011V13.86H10.45C10.7596 13.1218 11.2076 12.4497 11.7699 11.88H3.96011V9.9H11.8801V11.77C12.9381 10.699 14.3409 10.0366 15.8402 9.9V5.93996L9.9 0H1.98005C1.72004 0 1.46269 0.051222 1.22246 0.150726C0.982236 0.250231 0.763938 0.396084 0.580078 0.579944C0.396218 0.763804 0.250163 0.982035 0.150659 1.22226C0.0511549 1.46248 0 1.71997 0 1.97999V17.82C0 18.08 0.0511549 18.3374 0.150659 18.5777C0.250163 18.8179 0.396218 19.0362 0.580078 19.2201C0.763938 19.4039 0.982236 19.5498 1.22246 19.6493C1.46269 19.7488 1.72004 19.8 1.98005 19.8H10.7801C10.207 18.7943 9.90379 17.6575 9.9 16.5V15.84ZM8.91011 1.42999L14.3 6.92999H8.91011V1.42999Z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M16.5 11C15.0413 11 13.6425 11.5795 12.6111 12.6109C11.5796 13.6424 11 15.0413 11 16.5C11 17.9587 11.5796 19.3576 12.6111 20.3891C13.6425 21.4205 15.0413 22 16.5 22C17.9499 21.9718 19.3326 21.3833 20.3581 20.3579C21.3835 19.3325 21.9718 17.9499 22 16.5C22 15.0413 21.4207 13.6424 20.3892 12.6109C19.3578 11.5795 17.9587 11 16.5 11ZM19.25 17.05H17.05V19.25H15.95V17.05H13.75V15.95H15.95V13.75H17.05V15.95H19.25V17.05Z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
          <RNSVGDefs>
            <RNSVGClipPath
              name="clip0_464_280"
            >
              <RNSVGRect
                fill={4294967295}
                height="22"
                propList={
                  Array [
                    "fill",
                  ]
                }
                width="22"
                x={0}
                y={0}
              />
            </RNSVGClipPath>
          </RNSVGDefs>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
</RNCSafeAreaView>
`;
